### PR TITLE
[DEV-3484] Clear NAICS tree on reset search

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**

Forgot to deliver functionality to clear NAICs tree when user selects "reset search"

**Technical details:**

SubmitSearchButtonContainer now takes a new action to clear `redux.naics.checked` and `redux.naics.unchecked`

**JIRA Ticket:**
[DEV-3484](https://federal-spending-transparency.atlassian.net/browse/DEV-3484)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
`N/A` All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/containers/search/SearchSidebarSubmitContainer.jsx [hunk 1]
```diff
@@ -120,7 +123,13 @@ export default connect(
         stagedFilters: state.filters,
         appliedFilters: state.appliedFilters.filters
     }),
-    (dispatch) => bindActionCreators(combinedActions, dispatch)
+    (dispatch) => ({
+        ...bindActionCreators(combinedActions, dispatch),
+        resetNaicsTree: () => {
```
Comment (jonhill13) [inline src/js/containers/search/SearchSidebarSubmitContainer.jsx]: Do we want this resetNaicsTree here or should it be moved out to it's own component?

**Comment by jonhill13** [comment]
Also, @MaxwellKendall do we want this in QAT?

File: src/js/containers/search/SearchSidebarSubmitContainer.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/containers/search/SearchSidebarSubmitContainer.jsx]: Not sure what you mean here. Feel free to HMU on slack for a call if you wanna talk about it!

Comment (jonhill13) [reply src/js/containers/search/SearchSidebarSubmitContainer.jsx]: Yeah, I wrote that kinda fast.  Wouldn't we want `resetNaicsTree` in its own file or with the other naics actions in `redux/actions/`?

Or just use `setChecked` or `setUnchecked`

**Pull request merged**